tests/core/eth-module/test_transactions.py

```diff
@@ -193,6 +206,28 @@ def test_unmined_transaction_wait_for_receipt(w3):
     assert txn_receipt["blockHash"] is not None
 
 
+def test_eth_wait_for_transaction_receipt_transaction_indexing_in_progress(w3):
+    result_middleware = construct_result_generator_middleware(
+        {RPC.eth_getTransactionReceipt: lambda *_: {"status": 1}}
+    )
+    error_middleware = construct_error_generator_middleware(
+        {
+            RPC.eth_getTransactionReceipt: lambda *_: (
+                _swap_error_middleware_for_result_middleware_and_return_error(
+                    w3, result_middleware
+                )
+            )
+        }
+    )
+    w3.middleware_onion.inject(error_middleware, "error_middleware", layer=0)
+
+    receipt = w3.eth.wait_for_transaction_receipt(f"0x{'00' * 32}")
+    assert receipt == {"status": 1}
+
+    assert "error_middleware" not in w3.middleware_onion.middlewares
```

Comment:
Nit: This seems like it would always be true given the expected receipt came back. Curious if you have thoughts for this assertion?

Reply:
This is just to add an extra layer of assertion for better code readability. Sanity check that the `error_middleware` is getting cleaned up when we add the `result_middleware` which is why we don't remove it from the onion below.